### PR TITLE
btl_state_update_player_menu

### PR DESCRIPTION
**Pull request by Unnunu**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀 This PR matches 1 function (+0.28%) on `us`.

**Review by ethteck — APPROVED** [review]

File: src/415D90.c [hunk 1]
```diff
@@ -2381,48 +2370,47 @@ void btl_state_update_player_menu(void) {
...
             }
-            //904
+
             if (D_802ACC60 != 0) {
                 D_802ACC60--;
-            } else if (s00 != 0) {
-                set_animation(0, 0, 0x10004);
-                battleStatus->unk_4C[0] = battleStatus->currentSubmenu = battle_menu_submenuIDs[s00 - 1];
-                for (i = 0; i < 24; i++) {
+            } else if (submenuResult != 0){
```
Comment (nanaian) [inline src/415D90.c:2399]:
```suggestion
            } else if (submenuResult != 0) {
```

File: src/415D90.c [hunk 2]
```diff
@@ -2381,48 +2370,47 @@ void btl_state_update_player_menu(void) {
...
+                set_animation(ACTOR_PLAYER, 0, ANIM_Mario_Walking);
+                battleStatus->unk_4C[0] = battleStatus->currentSubmenu = battle_menu_submenuIDs[submenuResult - 1];
+                for (i = 0; i < ARRAY_COUNT(battleStatus->submenuMoves); i++) {
                     battleStatus->submenuMoves[i] = 0;
-                    battleStatus->submenuIcons[0] = 0;
+                    battleStatus->submenuIcons[0] = 0; ///< @bug ?
```
[the remark this answers is not in the document]
Comment (ethteck) [reply src/415D90.c:2404]: presumably because it should be using i as well

[the remark this answers is not in the document]
Comment (ethteck) [reply src/415D90.c:2404]: but idk actually

[the remark this answers is not in the document]
Comment (Unnunu) [reply src/415D90.c:2404]: Yes, it should be i. Although it doesn't seem to have any side effects.

**Review by nanaian — APPROVED** [review]

**Pull request merged**